tools/benchmark_suite_scheduler.py

```diff
@@ -0,0 +1,582 @@
+
+        self.validate()
+
+        # Will be set in #evaluate, once training_checkpoint_dir becomes available.
+        self.training_config = None
+        self.evaluation_results = None
+
+    def evaluate(self):
+        """
+        Evaluate the checkpoints. At a high level, this is the structure.
+
+        1. Load training YAML config file.
+        2. Monitor training checkpoints. When checkpoint is ready, launch the evaluation jobs.
+        3. Monitor the evaluation jobs. When evaluation jobs are complete, mark the results.
+        3. Once all jobs have been recorded, complete.
+        """
+        start_time = time.time()
+
+        # Wait for the training config to be available. This indicates the training has begun.
+        while True:
+            if time.time() - start_time > _TRAINING_CONFIG_WAIT_SECONDS:
+                raise RuntimeError(
+                    f"Training config still doesn't exist after:"
+                    f"{_TRAINING_CONFIG_WAIT_SECONDS / 60} minutes"
+                )
+
+            if PathManager.exists(self.training_checkpoint_file):
+                # Load training yaml config.
+                self._load_training_config()
+
+                # Build main training task in order to extract iteration info.
+                self.training_task = build_task(self.training_config)
+                self.training_task.dataloaders = self.training_task.build_dataloaders(
+                    False
+                )
+
+                # Generate evaluation results
+                self.evaluation_results = self._generate_initial_benchmark_results()
+                self._validate_evaluation_setup()
+
+                break
+
+            time.sleep(_SLEEP_TIME_SECONDS)
+
+        # Save initial evaluation benchmarks, for checkpointing reasons.
+        self.save_evaluation_benchmarks()
+
+        # Checkpoint folder is now available. Continuously monitor the training checkpoints,
+        # launch evaluation jobs as needed, monitor their progress, and record their results.
+        while True:
+            self._evaluate_checkpoints()
+
+            self._check_evaluation_jobs()
+
+            # Break if no more checkpoints to evaluate
+            if self._finished():
+                logging.info("Evaluations are finished")
+                break
+
+            time.sleep(_SLEEP_TIME_SECONDS)
+
+    def save_evaluation_benchmarks(self):
+        """
+        Create the /evaluations directory inside the training checkpoints dir.
+        Upload json file to the parent evaluation directories, as well as
+        to each child evaluation directories.
+        """
+        # Upload all checkpoints evaluations to parent checkpoint directory.
+        evaluation_dir = self.evaluation_dir()
+        parent_metrics_file = os.path.join(evaluation_dir, "evaluation_metrics.json")
+
+        if not PathManager.exists(evaluation_dir):
+            PathManager.mkdirs(evaluation_dir)
```

Comment:
nit: can we leverage this function directly https://github.com/facebookresearch/vissl/blob/master/vissl/utils/io.py#L122-L133 ?

Reply:
Nice. Thanks, I missed this one!